pkg/cluster/kubeconfig.go

```diff
@@ -0,0 +1,144 @@
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/submariner-io/armada/pkg/defaults"
+	"gopkg.in/yaml.v2"
+)
+
+type kubeConfig struct {
+	APIVersion string `yaml:"apiVersion"`
+	Clusters   []struct {
+		Cluster struct {
+			CertificateAuthorityData string `yaml:"certificate-authority-data"`
+			Server                   string `yaml:"server"`
+		} `yaml:"cluster"`
+		Name string `yaml:"name"`
+	} `yaml:"clusters"`
+	Contexts []struct {
+		Context struct {
+			Cluster string `yaml:"cluster"`
+			User    string `yaml:"user"`
+		} `yaml:"context"`
+		Name string `yaml:"name"`
+	} `yaml:"contexts"`
+	CurrentContext string `yaml:"current-context"`
+	Kind           string `yaml:"kind"`
+	Preferences    struct {
+	} `yaml:"preferences"`
+	Users []struct {
+		Name string `yaml:"name"`
+		User struct {
+			ClientCertificateData string `yaml:"client-certificate-data"`
+			ClientKeyData         string `yaml:"client-key-data"`
+		} `yaml:"user"`
+	} `yaml:"users"`
+}
+
+// PrepareKubeConfigs modifies kubconfig file generated by kind and returns the kubeconfig file path
+func PrepareKubeConfigs(clName string, sourceKubeConfigFilePath, masterIP string) error {
+	var kubeconf kubeConfig
+	currentDir, _ := os.Getwd()
+
+	_ = os.MkdirAll(defaults.LocalKubeConfigDir, os.ModePerm)
+	_ = os.MkdirAll(defaults.ContainerKubeConfigDir, os.ModePerm)
+	kindKubeFileName := strings.Join([]string{"kind-config", clName}, "-")
+	newLocalKubeFilePath := filepath.Join(currentDir, defaults.LocalKubeConfigDir, kindKubeFileName)
+	newContainerKubeFilePath := filepath.Join(currentDir, defaults.ContainerKubeConfigDir, kindKubeFileName)
+
+	sourceKubeFile, err := ioutil.ReadFile(sourceKubeConfigFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read kube config %s.", sourceKubeConfigFilePath)
+	}
+
+	err = yaml.Unmarshal(sourceKubeFile, &kubeconf)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read kube config %s.", sourceKubeConfigFilePath)
+	}
+
+	kubeconf.CurrentContext = clName
+	kubeconf.Contexts[0].Name = clName
+	kubeconf.Contexts[0].Context.Cluster = clName
+	kubeconf.Contexts[0].Context.User = clName
+	kubeconf.Clusters[0].Name = clName
+	kubeconf.Users[0].Name = clName
+
+	d, err := yaml.Marshal(&kubeconf)
+	if err != nil {
+		return errors.Wrapf(err, "failed to marshal kube config.")
+	}
+
+	err = ioutil.WriteFile(newLocalKubeFilePath, d, 0644)
+	if err != nil {
+		return errors.Wrapf(err, "failed to save kube config %s.", newLocalKubeFilePath)
+	}
+
+	kubeconf.Clusters[0].Cluster.Server = "https://" + masterIP + ":6443"
+	d, err = yaml.Marshal(&kubeconf)
+	if err != nil {
+		return errors.Wrapf(err, "failed to marshal kube config.")
+	}
+
+	err = ioutil.WriteFile(newContainerKubeFilePath, d, 0644)
```

Comment:
could you elaborate why you need to save the same file twice but with a change on `.Cluster.Server` ?